src/vm/stdinterfaces.cpp

```diff
@@ -575,30 +575,92 @@ ClassInfo_GetClassInfo(IUnknown* pUnk, ITypeInfo** ppTI)
         }
 
         MethodTable* pClass = pWrap->GetMethodTable();
-        IfFailThrow(GetITypeInfoForEEClass(pClass, ppTI, true/*bClassInfo*/));
+        IfFailThrow(GetITypeInfoForEEClass(pClass, ppTI, true /* bClassInfo */));
     }
     END_EXTERNAL_ENTRYPOINT;
 
     return hr;
 }
 
 //------------------------------------------------------------------------------------------
-// Helper to get the ITypeInfo* for a type.
-HRESULT GetITypeLibForEEClass(MethodTable *pClass, ITypeLib **ppTLB, int bAutoCreate, int flags)
+// Helper to get the ITypeLib* for a Assembly.
+HRESULT GetITypeLibForAssembly(_In_ Assembly *pAssembly, _Outptr_ ITypeLib **ppTlb)
 {
     CONTRACTL
     {
         NOTHROW;
         GC_TRIGGERS;
         MODE_PREEMPTIVE;
+        PRECONDITION(CheckPointer(pAssembly));
+        PRECONDITION(CheckPointer(ppTlb));
     }
     CONTRACTL_END;
 
-    return COR_E_NOTSUPPORTED;
-} // HRESULT GetITypeLibForEEClass()
+    // If the module wasn't imported from COM, fail. In .NET Framework the runtime
+    // would generate a ITypeLib instance, but .NET Core doesn't support that.
+    if (!pAssembly->IsImportedFromTypeLib())
+        return COR_E_NOTSUPPORTED;
 
+    HRESULT hr;
+
+    // Check for cached copy.
+    ITypeLib *pTlb = pAssembly->GetTypeLib();
+    if (pTlb != nullptr)
+    {
+        // If the cached value is the invalid sentinal, an attempt was already made but failed.
+        if (pTlb == Assembly::InvalidTypeLib)
+            return TLBX_E_LIBNOTREGISTERED;
+
+        // Return cached copy.
+        *ppTlb = pTlb;
+        return S_OK;
+    }
+
+    // Retrieve the guid for the assembly.
+    GUID assemblyGuid;
+    IfFailRet(GetTypeLibGuidForAssembly(pAssembly, &assemblyGuid));
+
+    // Retrieve the major and minor version number.
+    USHORT wMajor;
+    USHORT wMinor;
+    IfFailRet(GetTypeLibVersionForAssembly(pAssembly, &wMajor, &wMinor));
 
-HRESULT GetITypeInfoForEEClass(MethodTable *pClass, ITypeInfo **ppTI, int bClassInfo/*=false*/, int bAutoCreate/*=true*/, int flags)
+    // Attempt to load the exact TypeLib
+    hr = LoadRegTypeLib(assemblyGuid, wMajor, wMinor, &pTlb);
+    if (FAILED(hr))
+    {
+        // Try just the Assembly version
+        IfFailRet(pAssembly->GetVersion(&wMajor, &wMinor, nullptr, nullptr));
+        hr = LoadRegTypeLib(assemblyGuid, wMajor, wMinor, &pTlb);
+        if (FAILED(hr))
+        {
+            // Try loading the highest registered version.
+            hr = LoadRegTypeLib(assemblyGuid, -1, -1, &pTlb);
+            if (FAILED(hr))
+                pTlb = Assembly::InvalidTypeLib;
+        }
+    }
+
+    bool setCache = pAssembly->TrySetTypeLib(pTlb);
+    if (!setCache)
+    {
```

Comment:
Oops. Potential leak of `pTlb` here 😢

Reply:
I will wait for the current build to complete and then push the fix. It is a simple 2 line:

```
        if (pTlb != Assembly::InvalidTypeLib)
            pTlb->Release();
```

This builds find locally.